.github/actions/github-deployment-create/action.yml

```
@@ -0,0 +1,47 @@
+name: Create GitHub Deployment
+description: Create GitHub Deployment
+inputs:
+  github-token:
+    description: GitHub Token
+    required: true
+  environment:
+    description: The Environment for this Deployment
+    required: true
+  state:
+    description: The Initial State to set right after creation
+    default: ""
+
+outputs:
+  id:
+    description: The created deployment id
+    value: ${{ steps.create.outputs.id }}
+
+runs:
+  using: "composite"
+  steps:
+    - name: Create Deployment
+      id: create
+      env:
+        GH_TOKEN: ${{ inputs.github-token }}
+        TARGET_ENV: ${{ inputs.environment }}
+      shell: bash
+      run: |
+        URL="/repos/$GITHUB_REPOSITORY/deployments"
+
+        echo "Creating deployment at $URL for sha $GITHUB_SHA and env $TARGET_ENV"
+        # use a json object to be able to empty the required_contexts array
+        CREATED="$(jq -n --arg ref "$GITHUB_SHA" --arg environment "$TARGET_ENV" '{ref: $ref, environment: $environment, "auto_merge": false, "required_contexts": []}' | gh api -X POST $URL --input -)"
+        echo $CREATED
+
+        DEPLOYMENT_ID="$(echo $CREATED | jq -r ".id")"
+        echo "Created deployment $DEPLOYMENT_ID"
+
+        echo "id=$DEPLOYMENT_ID" >> $GITHUB_OUTPUT
+
+    - name: Update Deployment State
+      if: ${{ inputs.state != '' }}
+      uses: Alfresco/alfresco-build-tools/.github/actions/github-deployment-status-update@improvement/AAE-20957-deployment-status
```

```suggestion
      uses: Alfresco/alfresco-build-tools/.github/actions/github-deployment-status-update@v5.18.0
```

@gionn should I apply the following for the release script to be happy, please?

---

yes with [skip ci] otherwise the build will fail